src/test/suite/format/format.test.ts

```diff
@@ -31,6 +41,13 @@ const formatTests: FormatTest[] = [
     { filenameRoot: '189-space-at-end-of-line' },
     { filenameRoot: 'ahk-explorer' },
     { filenameRoot: 'demo' },
+    {
+        filenameRoot: 'indent-code-after-sharp-directive-false',
+        extensionOption: {
```

Comment:
Instead of a new `extensionOption` property, we should be able to refactor `options` to accept both `insertSpaces` and `indentCodeAfterSharpDirective` :)

Reply:
Add to `options` our new extension's setting and check it in `formatProvider.provideDocumentFormattingEdits()`:

if present in `options` -> use it, if not -> read global extension's setting?

Reply:
@kyklish try with the new "internalized" and "externalized" logic, it should be a lot easier now